<a href="https://colab.research.google.com/github/claregelbrugge/rush-sales-data/blob/main/RUSH_Case_Study_GB885.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#RUSH Sales Data Analysis

## Business objective:

You work as an analyst for RUSH, a globally renowned sportswear and footwear brand known for its innovative designs and performance-oriented products. The company stores its raw sales data as a collection of three tables:

*   TABLE_PRODUCTS
*   TABLE_RETAILERS
*   TABLE_SALES



This data contains the number of units sold, the total sales revenue, the location of the sales, type of product sold as well as other relevant information.

The VP of US Sales has tasked you with analyzing sales data for trends and insights that will help company leadership understand the market and opportunities for growth.

They have also asked you to answer several business questions.

### Load the data

In [1]:
# import pandas as pd
import pandas as pd

In [2]:
# load the three tables
products=pd.read_csv('TABLE_PRODUCTS_885.csv')
retailers=pd.read_csv('TABLE_RETAILER_885.csv')
sales=pd.read_csv('TABLE_SALES_885.csv')

In [3]:
# check that the product table loaded correctly
products.head(4)

,PRODUCT_ID|PRODUCT_NAME
0,020|Men's Street Footwear
1,030|Men's Athletic Footwear
2,120|Women's Street Footwear
3,130|Women's Athletic Footwear


In [4]:
# separate the columns
products = pd.read_csv("TABLE_PRODUCTS_885.csv", sep="|")

In [5]:
# check that it separated correctly
products.head()

,PRODUCT_ID,PRODUCT_NAME
0,20,Men's Street Footwear
1,30,Men's Athletic Footwear
2,120,Women's Street Footwear
3,130,Women's Athletic Footwear
4,40,Men's Apparel


### Inspect tables

In [6]:
# check the retailer table loaded correctly
retailers.head()

,RETAILER_ID,RETAILER,REGION,STATE,CITY
0,A00MOHCO,Amazon,Midwest,Ohio,Columbus
1,A00NMAPO,Amazon,Northeast,Maine,Portland
2,A00NMABO,Amazon,Northeast,Massachusetts,Boston
3,A00NNEMA,Amazon,Northeast,New Hampshire,Manchester
4,A00NVEBU,Amazon,Northeast,Vermont,Burlington


In [7]:
# check that the sales table loaded correctly
sales.head()

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD
0,1,A00MOHCO,1/1/2020,1,1,2020,20,50.0,1200,0.5,In-store
1,7,A00MOHCO,1/7/2020,1,7,2020,20,50.0,1250,0.5,In-store
2,13,A00MOHCO,1/25/2020,1,25,2020,20,50.0,1220,0.5,Outlet
3,19,A00MOHCO,1/31/2020,1,31,2020,20,50.0,1200,0.5,Outlet
4,25,A00MOHCO,2/6/2020,2,6,2020,20,60.0,1220,0.5,Outlet


In [8]:
# understand the data shape
products.shape

(6, 2)

In [9]:
# understand the data shape of retailers table
retailers.shape

(110, 5)

In [10]:
# understand the data shape of sales table
sales.shape

(9648, 11)

### Merge tables

In [11]:
# merge the data tables
sales_data = sales.merge(products, on='PRODUCT_ID', how='left')

In [12]:
# merge the data with retailer
sales_data = sales_data.merge(retailers, on='RETAILER_ID', how='left')

In [13]:
# check all the data together
sales_data.head(10)

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD,PRODUCT_NAME,RETAILER,REGION,STATE,CITY
0,1,A00MOHCO,1/1/2020,1,1,2020,20,50.0,1200,0.5,In-store,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
1,7,A00MOHCO,1/7/2020,1,7,2020,20,50.0,1250,0.5,In-store,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
2,13,A00MOHCO,1/25/2020,1,25,2020,20,50.0,1220,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
3,19,A00MOHCO,1/31/2020,1,31,2020,20,50.0,1200,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
4,25,A00MOHCO,2/6/2020,2,6,2020,20,60.0,1220,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
5,31,A00MOHCO,3/4/2020,3,4,2020,20,60.0,1250,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
6,37,A00MOHCO,3/10/2020,3,10,2020,20,60.0,1275,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
7,43,A00MOHCO,3/16/2020,3,16,2020,20,60.0,1250,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
8,49,A00MOHCO,4/19/2020,4,19,2020,20,60.0,1200,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus
9,57,A00MOHCO,4/27/2020,4,27,2020,20,65.0,1150,0.5,Outlet,Men's Street Footwear,Amazon,Midwest,Ohio,Columbus


In [14]:
# check that the merge worked
sales_data.shape

(10271, 16)

# EDA/Data Preparation:


In [15]:
# check for nulls
sales_data.isna().sum()

,0
ORDER_ID,0
RETAILER_ID,0
INVOICE_DATE,0
MONTH,0
DAY,0
YEAR,0
PRODUCT_ID,0
PRICE_PER_UNIT,2
UNITS_SOLD,0
OPERATING_MARGIN,0


In [16]:
# find the rows with missing prices
sales_data[sales_data['PRICE_PER_UNIT'].isna()]

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD,PRODUCT_NAME,RETAILER,REGION,STATE,CITY
98,591,A00NVEBU,4/2/2020,4,2,2020,20,NaN,525,0.35,In-store,Men's Street Footwear,Amazon,Northeast,Vermont,Burlington
99,597,A00NVEBU,4/8/2020,4,8,2020,20,NaN,525,0.50,In-store,Men's Street Footwear,Amazon,Northeast,Vermont,Burlington


In [17]:
# find the median for this specific product's price
sales_data.loc[sales_data['PRODUCT_ID'] == 20, 'PRICE_PER_UNIT'].median()

44.0

In [18]:
# replace missing prices with the median price for PRODUCT_ID 20
sales_data.loc[98, 'PRICE_PER_UNIT'] = 44
sales_data.loc[99, 'PRICE_PER_UNIT'] = 44

In [19]:
# check that the change was made
sales_data.loc[[98,99]]

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD,PRODUCT_NAME,RETAILER,REGION,STATE,CITY
98,591,A00NVEBU,4/2/2020,4,2,2020,20,44.0,525,0.35,In-store,Men's Street Footwear,Amazon,Northeast,Vermont,Burlington
99,597,A00NVEBU,4/8/2020,4,8,2020,20,44.0,525,0.50,In-store,Men's Street Footwear,Amazon,Northeast,Vermont,Burlington


The two missing price values were replaced with the median price for the corresponding product, which was $44.

In [20]:
# find the rows with missing retailer
sales_data[sales_data['RETAILER'].isna()]

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD,PRODUCT_NAME,RETAILER,REGION,STATE,CITY
1534,8668,999999999,7/23/2021,7,23,2021,20,60.0,298,0.42,Outlet,Men's Street Footwear,NaN,NaN,NaN,NaN


This item is missing retailer, region, state and city for this one and has a suspicious 999999 value for retailer_id.

In [21]:
# change the missing values to "Unknown"
sales_data.loc[1534, ['RETAILER_ID','RETAILER', 'REGION', 'STATE', 'CITY']] = 'Unknown'

In [22]:
# make sure it worked
sales_data.loc[1534]

,1534
ORDER_ID,8668
RETAILER_ID,Unknown
INVOICE_DATE,7/23/2021
MONTH,7
DAY,23
YEAR,2021
PRODUCT_ID,20
PRICE_PER_UNIT,60.0
UNITS_SOLD,298
OPERATING_MARGIN,0.42


The row contained a placeholder retailer ID (999999999) and missing retailer and location information. Because the correct values could not be determined, these fields were changed to "Unknown" rather than being guessed.

In [23]:
# check that no more remaining NAs
sales_data.isna().sum()

,0
ORDER_ID,0
RETAILER_ID,0
INVOICE_DATE,0
MONTH,0
DAY,0
YEAR,0
PRODUCT_ID,0
PRICE_PER_UNIT,0
UNITS_SOLD,0
OPERATING_MARGIN,0


In [24]:
# check for duplicated rows
sales_data.duplicated().sum()

np.int64(0)

There are no duplicated rows.

In [25]:
# check for data types
sales_data.dtypes

,0
ORDER_ID,int64
RETAILER_ID,object
INVOICE_DATE,object
MONTH,int64
DAY,int64
YEAR,int64
PRODUCT_ID,int64
PRICE_PER_UNIT,float64
UNITS_SOLD,object
OPERATING_MARGIN,float64


In [26]:
# convert invoice date to an actual date
sales_data['INVOICE_DATE'] = pd.to_datetime(sales_data['INVOICE_DATE'])

In [27]:
# convert units sold to a numeric data type
sales_data['UNITS_SOLD'] = pd.to_numeric(sales_data['UNITS_SOLD'], errors='coerce')

In [28]:
# check the data types again
sales_data.dtypes

,0
ORDER_ID,int64
RETAILER_ID,object
INVOICE_DATE,datetime64[ns]
MONTH,int64
DAY,int64
YEAR,int64
PRODUCT_ID,int64
PRICE_PER_UNIT,float64
UNITS_SOLD,float64
OPERATING_MARGIN,float64


In [29]:
# check for NAs again after data types are changed
sales_data.isna().sum()

,0
ORDER_ID,0
RETAILER_ID,0
INVOICE_DATE,0
MONTH,0
DAY,0
YEAR,0
PRODUCT_ID,0
PRICE_PER_UNIT,0
UNITS_SOLD,2
OPERATING_MARGIN,0


In [30]:
# check with describe to see if there are any major outliers
sales_data[['PRICE_PER_UNIT', 'UNITS_SOLD', 'OPERATING_MARGIN']].describe()

,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN
count,10271.000000,10269.000000,10271.000000
mean,54.573265,246.638232,0.426380
std,986.377876,211.881078,0.096018
min,7.000000,0.000000,0.100000
25%,35.000000,100.000000,0.350000
50%,45.000000,173.000000,0.420000
75%,55.000000,325.000000,0.500000
max,99999.000000,1275.000000,0.800000


A price_per_unit value value of 99,999 was identified as an extreme and likely erroneous value. We need to replace with the median price for the corresponding product ($44).

In [31]:
# find the item with 99999 price_per_unit value
sales_data[sales_data['PRICE_PER_UNIT'] == 99999]

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD,PRODUCT_NAME,RETAILER,REGION,STATE,CITY
423,2536,F00NNEMA,2021-05-23,5,23,2021,20,99999.0,520.0,0.4,Online,Men's Street Footwear,Foot Locker,Northeast,New Hampshire,Manchester


In [32]:
# replace this price per unit with the median used before
sales_data.loc[423, 'PRICE_PER_UNIT'] = 44

In [33]:
# see if it worked
sales_data.loc[423]

,423
ORDER_ID,2536
RETAILER_ID,F00NNEMA
INVOICE_DATE,2021-05-23 00:00:00
MONTH,5
DAY,23
YEAR,2021
PRODUCT_ID,20
PRICE_PER_UNIT,44.0
UNITS_SOLD,520.0
OPERATING_MARGIN,0.4


Looking at the description table, units_sold also has a minimum of 0, which means there are rows that are 0 or NA, which should be examined.

In [34]:
# check the missing values from units sold
sales_data[sales_data['UNITS_SOLD'].isna()]

,ORDER_ID,RETAILER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN,SALES_METHOD,PRODUCT_NAME,RETAILER,REGION,STATE,CITY
1021,6064,S00SALBI,2021-05-27,5,27,2021,20,51.0,NaN,0.45,Online,Men's Street Footwear,Sports Direct,South,Alabama,Birmingham
1527,8626,W00MIODE,2021-12-10,12,10,2021,20,29.0,NaN,0.46,Outlet,Men's Street Footwear,West Gear,Midwest,Iowa,Des Moines


Just keep the rows because no way to determine how many units sold.

In [35]:
# check describe again to see if everything looks good
sales_data.describe()

,ORDER_ID,INVOICE_DATE,MONTH,DAY,YEAR,PRODUCT_ID,PRICE_PER_UNIT,UNITS_SOLD,OPERATING_MARGIN
count,10271.000000,10271,10271.000000,10271.000000,10271.000000,10271.000000,10271.000000,10269.000000,10271.000000
mean,4970.630903,2021-05-13 12:00:54.678220288,6.456333,14.620290,2020.873235,80.001947,44.841495,246.638232,0.426380
min,1.000000,2020-01-01 00:00:00,1.000000,1.000000,2020.000000,20.000000,7.000000,0.000000,0.100000
25%,2568.500000,2021-02-18 00:00:00,3.000000,9.000000,2021.000000,30.000000,35.000000,100.000000,0.350000
50%,5136.000000,2021-06-05 00:00:00,6.000000,14.000000,2021.000000,120.000000,45.000000,173.000000,0.420000
75%,7236.000000,2021-09-17 00:00:00,9.000000,20.000000,2021.000000,130.000000,55.000000,325.000000,0.500000
max,9648.000000,2021-12-31 00:00:00,12.000000,31.000000,2021.000000,140.000000,110.000000,1275.000000,0.800000
std,2770.737186,NaN,3.461826,7.301233,0.332725,50.671441,14.789144,211.881078,0.096018


The max order_id is 9648, yet the count of order_ids is 10,271. This probably means that there are duplicated order_ids.

In [36]:
# check unique order_id
sales_data['ORDER_ID'].nunique()


9648

In [37]:
# check duplicated order_ids
sales_data['ORDER_ID'].duplicated().sum()

np.int64(623)

There are 623 repeated order_id values, but there are no completely duplicated rows. Repeated order IDs are likely due to orders containing multiple products, so these records were retained.

In [38]:
# look at unique product name values
sales_data['PRODUCT_NAME'].unique()

array(["Men's Street Footwear", "Men's Athletic Footwear",
       "Men's Apparel", "Women's Street Footwear",
       "Women's Athletic Footwear", "Women's Apparel"], dtype=object)

In [39]:
# look at unique retailer values
sales_data['RETAILER'].unique()

array(['Amazon', 'Foot Locker', "Kohl's", 'Sports Direct', 'Walmart',
       'West Gear', 'Unknown'], dtype=object)

In [40]:
# look at unique regions
sales_data['REGION'].unique()

array(['Midwest', 'Northeast', 'South', 'Southeast', 'West', 'Unknown'],
      dtype=object)

In [41]:
# look at unique states
sales_data['STATE'].unique()

array(['Ohio', 'Maine', 'Massachusetts', 'New Hampshire', 'Vermont',
       'Alabama', 'Kentucky', 'North Carolina', 'Alaska', 'Louisiana',
       'Georgia', 'Virginia', 'Arizona', 'Hawaii', 'Nebraska', 'Idaho',
       'Wyoming', 'Illinois', 'Iowa', 'Kansas', 'Michigan', 'Minnesota',
       'Missouri', 'North Dakota', 'South Dakota', 'Connecticut',
       'Delaware', 'Maryland', 'Florida', 'New York', 'Pennsylvania',
       'Rhode Island', 'South Carolina', 'West Virginia', 'Mississippi',
       'Tennessee', 'Texas', 'California', 'Washington', 'New Mexico',
       'New Jersey', 'Montana', 'Oklahoma', 'Arkansas', 'Colorado',
       'Nevada', 'Oregon', 'Utah', 'Indiana', 'Wisconsin', 'Unknown'],
      dtype=object)

In [42]:
# look at unique sales methods
sales_data['SALES_METHOD'].unique()

array(['In-store', 'Outlet', 'Ootlet', 'Online'], dtype=object)

In [43]:
# fix how outlet is spelled
sales_data['SALES_METHOD'] = sales_data['SALES_METHOD'].replace('Ootlet', 'Outlet')

# Analysis/Business Questions


Three of the business questions provided by the sales team require sales data, so we need to build a column for total sales calculated from price_per_unit and units_sold.

In [44]:
# calculate total sales revenue for each record
sales_data['SALES'] = sales_data['PRICE_PER_UNIT'] * sales_data['UNITS_SOLD']

In [45]:
# check the column calculation
sales_data[['PRICE_PER_UNIT', 'UNITS_SOLD', 'SALES']].head()

,PRICE_PER_UNIT,UNITS_SOLD,SALES
0,50.0,1200.0,60000.0
1,50.0,1250.0,62500.0
2,50.0,1220.0,61000.0
3,50.0,1200.0,60000.0
4,60.0,1220.0,73200.0


###Question 1: What product category (product) had the highest sales (in dollars) in 2021? How much did it sell?

In [46]:
# create new sales variable for 2021
sales_2021 = sales_data[sales_data['YEAR'] == 2021]

# group by product name and take sum sales from each product category
product_sales = sales_2021.groupby('PRODUCT_NAME')['SALES'].sum()

# get the product name with the most sales using ascending
product_sales.sort_values(ascending=False).head(1).round(2)

,SALES
PRODUCT_NAME,
Men's Street Footwear,23298116.0


**Men's street footwear** had the highest sales, with approximately **$23.2 million** in sales.

###Question 2: What state had the highest sales (in dollars) of women's products in 2021? How much was it?

In [47]:
# check how product names are labelled for each sex
sales_data['PRODUCT_NAME'].unique()

array(["Men's Street Footwear", "Men's Athletic Footwear",
       "Men's Apparel", "Women's Street Footwear",
       "Women's Athletic Footwear", "Women's Apparel"], dtype=object)

In [48]:
# create variable for women's category in 2021
women_2021 = sales_data[
    (sales_data['YEAR'] == 2021) &
    (sales_data['PRODUCT_NAME'].str.contains("Women's", na=False))
]

# group the 2021 women's sales data together
women_state_sales = women_2021.groupby('STATE')['SALES'].sum()

# see the highest sales value
women_state_sales.sort_values(ascending=False).head(1)

,SALES
STATE,
Maine,2176301.0


**Maine** was the state with the highest women's product sales of **2.1 million**.

###Question 3: What state had the highest sales (in dollars) of men's products in 2021? How much was it?

In [49]:
# make a variable for men's category in 2021
men_2021 = sales_data[
    (sales_data['YEAR'] == 2021) &
    (sales_data['PRODUCT_NAME'].str.contains("Men's", na=False))
]

# group the 2021 men's sales data together
men_state_sales = men_2021.groupby('STATE')['SALES'].sum()

# get the highest sales value of the categories
men_state_sales.sort_values(ascending=False).head(1)

,SALES
STATE,
Delaware,2334300.0


**Delaware** was the state with the highest men's product sales of **2.3 million**.

###Question 4: What retailer purchased the most units in 2021? In 2020?

In [50]:
# make a variable for 2021 units purchased
retailer_units_2021 = sales_data[
    sales_data['YEAR'] == 2021
].groupby('RETAILER')['UNITS_SOLD'].sum()

# sort the values in descending order
retailer_units_2021.sort_values(ascending=False).head(1)

,UNITS_SOLD
RETAILER,
Foot Locker,1097410.0


In [51]:
# do the same for 2020
retailer_units_2020 = sales_data[
    sales_data['YEAR'] == 2020
].groupby('RETAILER')['UNITS_SOLD'].sum()

# sort the data in descending order
retailer_units_2020.sort_values(ascending=False).head(1)

,UNITS_SOLD
RETAILER,
Amazon,317930.0


**Foot Locker** was the retailer with the most units sold in 2021, and **Amazon** was the retailer with the most units sold in 2020.

##Additional Insights

#### Q: What were the sales by year? Did sales decrease or increase?

In [52]:
# group the sales data by year and sum up all sales
sales_data.groupby('YEAR')['SALES'].sum()

,SALES
YEAR,
2020,24220525.0
2021,98174537.0


Sales in 2020 were 24.2 million while sales in 2021 were 98.1 million. So 2021 had significantly more sales.

In [63]:
# use group by to get sales data by year
yearly_sales = sales_data.groupby('YEAR')['SALES'].sum()

# calculate YoY growth
growth = (
    (yearly_sales[2021] - yearly_sales[2020])
    / yearly_sales[2020]
) * 100

growth

np.float64(305.3361229783417)

Sales increased 305% YoY from 2020 to 2021.

#### Q: How did sales change over the months?

In [53]:
# pull monthly sales by grouping by month and year summed
monthly_sales = sales_data.groupby(
    ['YEAR', 'MONTH']
)['SALES'].sum()

monthly_sales

YEAR  MONTH
2020  1         2312746.0
      2         2140813.0
      3         2474202.0
      4         3176281.0
      5         2164764.0
      6         1084194.0
      7         2182388.0
      8         2641630.0
      9         2367686.0
      10        1428569.0
      11        1168050.0
      12        1079202.0
2021  1         7580341.0
      2         6256659.0
      3         5307200.0
      4         6595521.0
      5         8734090.0
      6         8890823.0
      7        10657331.0
      8         9897471.0
      9         8163271.0
      10        7236980.0
      11        8104414.0
      12       10750436.0
Name: SALES, dtype: float64

In [54]:
# view for what month generated the highest sales overall, not including year
sales_data.groupby('MONTH')['SALES'].sum().sort_values(ascending=False)

,SALES
MONTH,
7,12839719.0
8,12539101.0
12,11829638.0
5,10898854.0
9,10530957.0
6,9975017.0
1,9893087.0
4,9771802.0
11,9272464.0


July generated the highest combined sales across 2020 and 2021.

#### Q: What were the sales per retailer?

In [55]:
# use group by to look at sales per retailer
retailer_sales = sales_data.groupby('RETAILER')['SALES'].sum()

# sort the values in descending order
retailer_sales.sort_values(ascending=False)

,SALES
RETAILER,
Foot Locker,53643305.0
Amazon,28008825.0
West Gear,14584306.0
Sports Direct,13031410.0
Kohl's,10852813.0
Walmart,2256523.0
Unknown,17880.0


Foot Locker generated the highest total sales, with approximately 53.6 million in revenue.

In [56]:
# make a group by function to sort by year and retailer
sales_data.groupby(
    ['YEAR', 'RETAILER']
)['SALES'].sum().sort_values(ascending=False)

YEAR  RETAILER     
2021  Foot Locker      53643305.0
2020  Amazon           17521350.0
2021  West Gear        12405097.0
      Sports Direct    12130175.0
      Amazon           10487475.0
      Kohl's            7234082.0
2020  Kohl's            3618731.0
2021  Walmart           2256523.0
2020  West Gear         2179209.0
      Sports Direct      901235.0
2021  Unknown             17880.0
Name: SALES, dtype: float64

#### Q: What are the sales per region?

In [57]:
# create a variable for region sales
region_sales = sales_data.groupby('REGION')['SALES'].sum()

region_sales.sort_values(ascending=False)

,SALES
REGION,
Northeast,39316504.0
Midwest,31433570.0
West,24820550.0
Southeast,15356136.0
South,11450422.0
Unknown,17880.0


The Northeast seemed to generate the most sales, followed by the Midwest. The South generated the least amount of sales.

#### Q: What sales method generated the most sales dollars?

In [58]:
sales_data.groupby('SALES_METHOD')['SALES'].sum().sort_values(ascending=False)

,SALES
SALES_METHOD,
Online,46965132.0
Outlet,39782355.0
In-store,35647575.0


Online generated the most followed by outlet.

#### Q: Which products or retailers generate the highest operating margins?

In [59]:
# group by product name and operating margin mean
sales_data.groupby('PRODUCT_NAME')['OPERATING_MARGIN'].mean().sort_values(
    ascending=False
)

,OPERATING_MARGIN
PRODUCT_NAME,
Men's Street Footwear,0.448506
Women's Apparel,0.444352
Women's Athletic Footwear,0.427538
Men's Apparel,0.416850
Women's Street Footwear,0.414019
Men's Athletic Footwear,0.406984


Men's Street Footwear had the highest operating margin at 45%.

In [61]:
# group by retailer and operating margin
sales_data.groupby('RETAILER')['OPERATING_MARGIN'].mean().sort_values(
    ascending=False
)

,OPERATING_MARGIN
RETAILER,
Sports Direct,0.496944
Walmart,0.475141
Kohl's,0.436816
West Gear,0.435817
Unknown,0.420000
Foot Locker,0.367368
Amazon,0.357534


Sports Direct had the highest average operating margin at 50%, closely followed by Walmart at 47.5%.